petab/composite_problem.py

```diff
@@ -0,0 +1,75 @@
+"""PEtab problems consisting of multiple models"""
+import os
+from typing import List, Union, Dict
+
+import pandas as pd
+
+from . import parameters
+from . import problem
+from . import yaml
+
+
+class CompositeProblem:
+    """Representation of a PEtab problem consisting of multiple models
+
+    Attributes:
+        problems:
+            List ``petab.Problems``
+        parameter_df:
+            PEtab parameter DataFrame
+    """
+
+    def __init__(
+            self,
+            parameter_df: pd.DataFrame = None,
+            problems: List[problem.Problem] = None):
+        """Constructor
+
+        Arguments:
+            parameter_df:
+                see CompositeProblem.parameter_df
+            problems:
+                see CompositeProblem.problems
+        """
+        self.problems: List[problem.Problem] = problems
+        self.parameter_df: pd.DataFrame = parameter_df
+
+    @staticmethod
+    def from_yaml(yaml_config: Union[Dict, str]
+                  ) -> 'CompositeProblem':
+        """Create from YAML file
+
+        Factory method to create a CompositeProblem instance from a PEtab
+        YAML config file
+
+        Arguments:
+            yaml_config: PEtab configuration as dictionary or YAML file name
+        """
+
+        old_wd = os.getcwd()
+        if isinstance(yaml_config, str):
+            new_wd = os.path.dirname(yaml_config)
+            yaml_config = yaml.load_yaml(yaml_config)
+
+        yaml_config = yaml.load_yaml(yaml_config)
+
+        try:
+            if new_wd:
+                os.chdir(new_wd)
```

Comment:
do you have to chdir? can potentially lead to problems e.g. when stuff fails, or when log files are written. can this be done by prefixing the locations instead?

Reply:
Probably rather exotic use case where this causes trouble (changing back at end of function) but will adapt.

Reply:
doesn't seem adapted yet, but should be fine